arangod/Cluster/ClusterInfo.cpp

```diff
@@ -3628,9 +3835,33 @@ Result ClusterInfo::dropCollectionCoordinator(  // drop collection
     return Result(TRI_ERROR_NO_ERROR);
   }
 
+  // Delete replicated states in case we are using Replication2
+  auto replicatedStatesCleanup = futures::Future<Result>{std::in_place};
+  if (coll->replicationVersion() == replication::Version::TWO) {
+    std::vector<replication2::LogId> stateIds;
+    for (auto pair : VPackObjectIterator(shardsSlice)) {
+      auto shardId = pair.key.copyString();
+      auto stateId = std::string_view{shardId.substr(1, shardId.size() - 1)};
+      auto logId = replication2::LogId::fromString(stateId);
+      ADB_PROD_ASSERT(logId.has_value())
+          << " converting " << stateId
+          << " to LogId failed, during replicated state deletion for shard "
+          << shardId;
```

Comment:
Couldn't we use `LogicalCollection::shardIdToLogId` here?

Reply:
Yes, that would be nice.